clang/test/CodeGenSYCL/save-user-names.cpp

```diff
@@ -0,0 +1,74 @@
+// RUN:  %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+// Test to verify that user specified names are retained in openCL
```

Comment:
@bader, will it be more correct if we call it a SPIR kernel?
```suggestion
// Test to verify that user specified names are retained in SPIR
```